Feat: Post 도메인 이미지 업로드 구현 및 테스트 코드 구현

kamillee0918 opened the pull request:
## ⭐Key Changes
> 핵심 변경 사항에 대해서 적어주세요.
1. 포스트 추가 / 조회 / 수정 / 삭제 관련 테스트 코드를 추가했어요...

<br>

## 📌Issue
> 닫을 issue 번호 : resolved #30

<br>

## 🩼Branch
> 반영 branch feat-30 -> dev

<br>

## 👪To Reviewers
> reviewer에게 하고 싶은 말을 적어주세요.
* 코드 리뷰 부탁드려요...

kamillee0918 committed:
Feat: Post 도메인에 이미지 업로드 코드 구현

### 구현 사항
1. 이미지 업로드 및 포스트 조회 시 이미지 URL을 포함하도록 코드를 수정했어요.

kamillee0918 committed:
Feat: Post 도메인에 이미지 업로드 코드 구현 #24

kamillee0918 committed:
Merge remote-tracking branch 'origin/feat-24' into feat-24

kamillee0918 committed:
Add: oauth2 패키지 추가 #24

### 추가 사항
1. 누락되어 있었던 oauth2 패키지를 다시 추가했어요. 변경된 코드는 없어요...

kamillee0918 committed:
Add: oauth2 패키지 추가

### 추가 사항
1. 누락되어 있었던 oauth2 패키지를 다시 추가했어요. 변경된 코드는 없어요...

kamillee0918 committed:
Revert "Add: oauth2 패키지 추가"

This reverts commit e791340884d906a0927f99eff06cb3fa4cb149fe.

kamillee0918 committed:
Merge remote-tracking branch 'origin/feat-24' into feat-24

kamillee0918 committed:
Feat: Post 도메인 테스트 코드 추가

### 추가 사항
1. 포스트 추가 / 조회 / 수정 / 삭제에 관한 테스트 코드를 추가했어요.
2. 기존 DTO에서 삭제 전용 생성자를 추가했어요.

kamillee0918 committed:
Merge remote-tracking branch 'origin/dev' into feat-30

kamillee0918 committed:
Refactor: Post 도메인 이미지 업로드 로직 개선

### 수정 사항
1. 테스트 코드를 포함하여 포스트 생성/수정/삭제 로직을 리팩토링했어요..

kamillee0918 commented:
Post 도메인, 이미지 업로드 로직을 개선했어요..

수정이 필요한 부분이 있으면 말씀해 주세요.. ㅠ

Hexeong commented on src/main/java/com/noblesse/backend/post/api/command/PostCommandController.java:
```diff
@@ -5,50 +5,99 @@
     public ResponseEntity<Long> createPost(
-            @RequestBody PostDTO command
+            @RequestParam("postTitle") String postTitle,
+            @RequestParam("userId") Long userId,
```
form data로 userId를 넘기는 방법은 보안상 안좋을 것 같아서 해당 유저가 가지고 있는 토큰에서 userId를 추출하는 방식으로 가져와야 될 것 같습니다..!

Hexeong commented on src/main/java/com/noblesse/backend/post/api/command/PostCommandController.java:
```diff
@@ -5,50 +5,99 @@
     public ResponseEntity<Void> deletePost(
             @PathVariable("postId") Long postId,
-            @RequestBody PostDTO command
+            @RequestParam("userId") Long userId
```
여기 또한 토큰에서 userId 추출하는 방식으로 부탁드립니다

Hexeong reviewed (changes requested):
userId를 사용하는 방식만 고치면 될 것 같습니다 :)

kamillee0918 committed:
Merge remote-tracking branch 'origin/dev' into feat-30

kamillee0918 committed:
Refactor: Post 도메인 이미지 업로드 로직 개선

### 수정 사항
1. 헤더에 로그인한 사용자의 토큰에서 `userId`를 추출하는 방식으로 코드를 수정했어요..! 확인이 늦어서 죄송합니다..! 잘못 작성되었거나, 다른 좋은 방식이 있으면 말씀해 주세요!

kamillee0918 commented:
말씀하신 요구 사항에 충족(?)되는 코드를 리팩토링 했어요..! 코드 리뷰 부탁드립니다.. :)

Hexeong reviewed (approved):
LGTM :)
